src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
>Only reliably for the first 2 weeks of uptime. After that QPC can return numbers that can't fit in a double without losing precision.

How so? Keep in mind, that what matters is not the absolute value returned by QPC. What matters is the clock resolution, relative to the precision of the representation. QPC has a resolution of 1 microsecond or better. A 52-bit mantissa can represent 2^52 microseconds without losing precision with respect to the internal clock resolution. That's over 2 million years.